src/Microsoft.AspNetCore.Rewrite/Internal/ConditionCollection.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Microsoft.AspNetCore.Rewrite.Internal
+{
+    public class ConditionCollection : IEnumerable<Condition>
+    {
+        private readonly List<Condition> _conditions = new List<Condition>();
+
+        public enum ConditionGrouping
+        {
+            And,
+            Or
+        }
+
+        public ConditionGrouping Grouping { get; }
+        public bool TrackAllCaptures { get; }
+
+        public ConditionCollection()
+            :this(ConditionGrouping.And, trackAllCaptures: false)
+        {
+        }
+
+        public ConditionCollection(ConditionGrouping grouping, bool trackAllCaptures)
+        {
+            Grouping = grouping;
+            TrackAllCaptures = trackAllCaptures;
+        }
+
+        public ConditionCollection(ConditionGrouping grouping, bool trackAllCaptures, Condition condition)
```

Comment:
I don't think the 3rd and 4th constructor overloads are being used anywhere. Remove.